### ISSUE TITLE
invalid flag: --unpack on ADD

### ISSUE THREAD

**Issue by 0xDagal** [hadolint/hadolint#1145]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

Expected behavior

When using

```
ADD --unpack=
```

I am getting invalid flag: --unpack.
Actual behavior

hadolint should be able to parse this flag.
Steps to reproduce the behavior

See above,.

Output of `hadolint --version`:

`Haskell Dockerfile Linter 2.14.0`

**Comment by tib-rijo** [comment]
Continues to date:

```sh
podman pull hadolint/hadolint
podman run --rm -v ${PWD}/context:/context:z hadolint/hadolint hadolint /context/stable/Dockerfile       
# /context/stable/Dockerfile:146:19 unexpected '-' expecting '\', at least one space, or only the --checksum, --chown, --chmod, --link, --exclude flags or the src and dest paths
podman run --rm -v ${PWD}/context:/context:z hadolint/hadolint hadolint --version                    
# Haskell Dockerfile Linter 2.14.0
```

Since it is no violation of no rule, it can also not be ignored with global or local rules, nor omitted with ignoring any failures with `-t` / `--faulure-threshold`.

```sh
$ podman run --rm -v ${PWD}/context:/context:z hadolint/hadolint hadolint -t none /context/stable/Dockerfile  
/context/stable/Dockerfile:146:19 unexpected '-' expecting '\', at least one space, or only the --checksum, --chown, --chmod, --link, --exclude flags or the src and dest paths
$ echo $?           
1
```

The option `--no-fail` allows a workaround.

```sh
$ podman run --rm -v ${PWD}/context:/context:z hadolint/hadolint hadolint --no-fail /context/stable/Dockerfile
/context/stable/Dockerfile:146:19 unexpected '-' expecting '\', at least one space, or only the --checksum, --chown, --chmod, --link, --exclude flags or the src and dest paths
$ echo $?                                                                                                     
0
```

`hadolint` output is then only of informative nature.

**Comment by m-ildefons** [comment]
Fixed in v2.15.0
